subprojects/plugins/src/integTest/groovy/org/gradle/api/plugins/JvmTestSuitePluginIntegrationTest.groovy

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.ToBeFixedForConfigurationCache
+
+class JvmTestSuitePluginIntegrationTest extends AbstractIntegrationSpec {
+
+    @ToBeFixedForConfigurationCache(because = ":outgoingVariants")
+    def "JVM Test Suites plugin adds outgoing variants for default test suite"() {
+        buildFile << """
+            plugins {
+                id 'jvm-test-suite'
+                id 'java'
+            }
+            """
+
+        file("src/test/java/SomeTest.java") << """
+            import org.junit.Test;
+
+            public class SomeTest {
+                @Test public void foo() {}
+            }
+            """.stripIndent()
+
+        expect:
+        succeeds "outgoingVariants"
+
+        outputContains("""
+            --------------------------------------------------
+            Variant testDataElementsForTest
+            --------------------------------------------------
+            Capabilities
+                - :${getTestDirectory().getName()}:unspecified (default capability)
+            Attributes
+                - org.gradle.category      = documentation
+                - org.gradle.docstype      = test-results-bin
+                - org.gradle.targetname    = test
+                - org.gradle.testsuitename = test
+                - org.gradle.testsuitetype = unit-tests
+                - org.gradle.usage         = verification
+
+            Artifacts
+                - build${File.separator}test-results${File.separator}test${File.separator}binary${File.separator}results.bin (artifactType = binary)
```

Comment:
Ack, these things are making me go cross-eyed.

WDYT about something like this to improve readability?
https://github.com/gradle/gradle/commit/b1be20f871f819bdd94c5a2c7b94223105292df9